lib/std/zig/system/NativeTargetInfo.zig

```diff
@@ -142,7 +142,10 @@ pub fn detect(cross_target: CrossTarget) DetectError!NativeTargetInfo {
                     error.Unexpected => return error.OSVersionDetectionFail,
                 };
 
-                if (std.SemanticVersion.parse(buf[0 .. len - 1])) |ver| {
+                buf[len - 1] = '.';
+                buf[len] = '0';
```

Comment:
this will overwrite data from the sysctl call if the sysctl call fills up the buffer.

Reply:
Do you prefer something like:
```

if ( len + 2 < 64 ){
    buf[len - 1] = '.';
    buf[len] = '0';
} else {
   new_buf  = buf + ".0";    // because buf is a 64 array;
}

```

Reply:
That looks needlessly complex. I'd pass `buf[0..buf.len - 2]` to the sysctl instead.

Reply:
"I'd pass buf[0..buf.len - 2] to the sysctl instead."
 
idk what you mean by "pass `buf[0..buf.len - 2]` to the sysctl" because we use sysctl to obtain version buffer/version buffer length, that's  explaining why buf and len are passed by reference;

I had to give ".0" at end of buf(="7.3") for satisfied the parser of `std.SemanticVersion` where "7.3" doesn't work. That's were the issue come from.  Also idk what do in the else if buf fills up i can't write new character anymore on it.